torch/_prims/executor.py

```diff
@@ -45,34 +45,52 @@ def call_function(self, target, args, kwargs):
                     args = tuple(map(_to_nvfuser_constant, args))
                     target = target.impl_nvfuser
                     args = (fd,) + args
-                    return target(*args, **kwargs)
+                    nv_args = [to_nv(x) for x in args]
+                    nv_kwargs = {k: to_nv(v) for k, v in kwargs.items()}
+                    return target(*nv_args, **nv_kwargs)
 
             def to_nv(arg):
                 if isinstance(arg, torch.Tensor):
                     x = fd.define_tensor(
                         arg.size(), arg.stride(), getnvFuserDtype(arg.dtype)
                     )
-                    fd.add_input(x)
                     return x
+                elif isinstance(arg, Number):
+                    return fd.define_constant(arg)
                 else:
                     return arg
 
+            def add_inputs(arg):
+                x = to_nv(arg)
+                if isinstance(x, TensorView):
+                    fd.add_input(x)
+                return x
+
             # Transforms graph to call nvfuser lowerings
-            nv_args = tree_map(to_nv, args)
-            nv_kwargs = tree_map(to_nv, kwargs)
+            nv_args = tree_map(add_inputs, args)
+            nv_kwargs = tree_map(add_inputs, kwargs)
 
             out = FusionInterpreter(gm).run(*nv_args, **nv_kwargs)
             flat_out, unflatten_spec = torch.utils._pytree.tree_flatten(out)
             for o in flat_out:
-                fd.add_output(o)
+                if isinstance(o, TensorView) or isinstance(o, Val):
+                    fd.add_output(o)
 
-            return torch.utils._pytree.tree_unflatten(
-                fusion.execute(
-                    tuple(arg for arg in args if isinstance(arg, torch.Tensor))
-                ),
-                unflatten_spec,
+            nv_results = fusion.execute(
+                tuple(arg for arg in args if isinstance(arg, torch.Tensor))
             )
 
+            results = []
+            i = 0
+            for o in flat_out:
+                if isinstance(o, TensorView) or isinstance(o, Val):
+                    results.append(nv_results[i])
+                    i += 1
+                else:
+                    results.append(o)
```

Comment:
Could you please clarify what function requires these changes?

Reply:
This is used to handle constant tensors in the output, such as the non-floating and non-complex case of this ref:
```python
@_make_elementwise_unary_reference(ELEMENTWISE_TYPE_PROMOTION_KIND.ALWAYS_BOOL)
def isposinf(a: TensorLikeType) -> TensorLikeType:
    if a.dtype.is_complex:
        msg = f"Complex dtype is not supported for isposinf, got dtype {a.dtype}"
        raise ValueError(msg)
    elif a.dtype.is_floating_point:
        return eq(a, float("inf"))
    return zeros_like(a, dtype=torch.bool)
```
where for this case, it is just a `return constant_tensor`.

But as you have pointed out in https://github.com/pytorch/pytorch/pull/78655/files#r899877833, this is not sufficient to handle constant tensors for all cases.